ui/app/src/state/reducers/pathNavigatorTree.ts

```diff
@@ -77,10 +77,17 @@ const restoreTree = (state, payload) => {
   const childrenByParentPath = chunk.childrenByParentPath;
   const totalByPath = chunk.totalByPath;
   const offsetByPath = chunk.offsetByPath;
-  items.forEach((item) => {
-    totalByPath[item.path] = item.childrenCount;
-  });
+  // Set totalByPath of items that are not in the children object (items not expanded) so we can display the
+  // expand/collapse icon (if they have children). We filter out the items not in the children object to prevent displaying
+  // an incorrect total count (since the children object has the total considering filters).
+  items
+    .filter((item) => !children[item.path])
+    .forEach((item) => {
+      totalByPath[item.path] = item.childrenCount;
+    });
   Object.keys(children).forEach((parentPath) => {
+    // Initialize the childrenByParentPath with an empty array.
+    childrenByParentPath[parentPath] = [];
```

Comment:
Why? Explain. The comment currently says what's evident by simply reading the code; it needs the rationale.
```suggestion
    // Initialize the childrenByParentPath with an empty array so that blah blah...
    childrenByParentPath[parentPath] = [];
```